src/Control/Lens/Fold.hs

```diff
@@ -1951,19 +1960,25 @@ ipre l = dimap (getFirst . getConst #. l (Indexed $ \i a -> Const (First (Just (
 ------------------------------------------------------------------------------
 
 -- | Retrieve the first value targeted by a 'Fold' or 'Traversal' (or 'Just' the result
--- from a 'Getter' or 'Lens'). See also ('^?').
+-- from a 'Getter' or 'Lens'). See also 'firstOf' and '^?', which are similar with
+-- some subtle differences (explained below).
 --
 -- @
 -- 'Data.Maybe.listToMaybe' '.' 'toList' ≡ 'preview' 'folded'
 -- @
 --
--- This is usually applied in the 'Control.Monad.Reader.Reader'
--- 'Control.Monad.Monad' @(->) s@.
+-- Unlike '^?', this function uses a 
+-- 'Control.Monad.Reader.MonadReader' to read the value to be focused in on.
+-- This allows one to pass the value as the last argument by using the 
+-- 'Control.Monad.Reader.MonadReader' instance for @(->) s@:
 --
 -- @
 -- 'preview' = 'view' '.' 'pre'
 -- @
 --
+-- In this case, it can be helpful to think of 'preview' as having one of these
```

Comment:
We don't state this. Many combinators in lens have specialised types **last** in the documentation. For example
http://hackage.haskell.org/package/lens-4.17/docs/Control-Lens-Traversal.html#v:traverseOf